SU2_CFD/src/integration/CMultiGridIntegration.cpp

```diff
@@ -692,6 +692,29 @@ void CMultiGridIntegration::NonDimensional_Parameters(CGeometry **geometry, CSol
       solver_container[FinestMesh][FLOW_SOL]->Momentum_Forces(geometry[FinestMesh], config);
       solver_container[FinestMesh][FLOW_SOL]->Friction_Forces(geometry[FinestMesh], config);
 
+      /*--- Calculate the turbo performance ---*/
+      if (config->GetBoolTurbomachinery()){
+
+        /*--- Average quantities at the inflow and outflow boundaries ---*/
+
+        solver_container[FinestMesh][FLOW_SOL]->TurboAverageProcess(solver_container[FinestMesh], geometry[FinestMesh],config,INFLOW);
+        solver_container[FinestMesh][FLOW_SOL]->TurboAverageProcess(solver_container[FinestMesh], geometry[FinestMesh], config, OUTFLOW);
+
+        /*--- Gather Inflow and Outflow quantities on the Master Node to compute performance ---*/
+
+        solver_container[FinestMesh][FLOW_SOL]->GatherInOutAverageValues(config, geometry[FinestMesh]);
+
+        /* --- compute turboperformance for each stage and the global machine ---*/
+        //TODO: for multi-zone turbo this should be move to the last zone ---*/
+          solver_container[FinestMesh][FLOW_SOL]->ComputeTurboPerformance(config, geometry[FinestMesh]);
+      }
```

Comment:
Could these be wrapped by a single method? Just to reduce the number of virtual methods in CSolver (less virtual boilerplate to maintain)